pkg/resources/metrics.go

```diff
@@ -170,6 +171,24 @@ func reconcilePrometheusRule(ctx context.Context, client k8sclient.Client, ruleN
 
 	// create or update the resource
 	_, err := controllerutil.CreateOrUpdate(ctx, client, rule, func() error {
+		rule.Name = ruleName
+		rule.Namespace = ns
+		rule.Spec.Groups = []prometheusv1.RuleGroup{
```

Comment:
is it possible there are multiple groups, and if so, would it make it easier to overwrite the entirety of `spec.groups` instead of overriding individual fields in 1 group?

Reply:
:eyes: Think I tried something like that earlier but didn't appear to update.

Reply:
just testing 
```
		rule.Spec = prometheusv1.PrometheusRuleSpec{groups}
```

Reply:
From what I can see debugging the `groups` object doesn't update inside the controllerutil.CreateOrUpdate block but does outside of that block, not sure why, will revert to original method for updating